services/venice-controller/src/main/java/com/linkedin/venice/controller/StoreBackupVersionCleanupService.java

```diff
@@ -232,13 +242,14 @@ private boolean validateAllServerOnCurrentVersion(Store store, String clusterNam
    * @return whether any backup version is removed or not
    */
   protected boolean cleanupBackupVersion(Store store, String clusterName) {
-    if (!whetherStoreReadyToBeCleanup(store, defaultBackupVersionRetentionMs, time)) {
+    int currentVersion = store.getCurrentVersion();
+
+    if (!whetherStoreReadyToBeCleanup(store, defaultBackupVersionRetentionMs, time, currentVersion)) {
       // not ready to clean up backup versions yet
       return false;
     }
 
     List<Version> versions = store.getVersions();
-    int currentVersion = store.getCurrentVersion();
 
     // Do not delete version unless all routers and all servers are on same current version
     if (multiClusterConfig.getControllerConfig(clusterName).isBackupVersionMetadataFetchBasedCleanupEnabled()
```

Comment:
How can we enable this feature by default because of the new repush cleanup logic?
Otherwise, we need to delay the deletion of the source version for a while to avoid race conditions.
I guess it is good to add a delay regardless, such as 1h?

Reply:
Sure, having some wait time regardless is good idea.